influxdb/src/main/java/com/hazelcast/jet/influxdb/InfluxDbSources.java

```diff
@@ -0,0 +1,343 @@
+                                                     @Nonnull SupplierEx<InfluxDB> connectionSupplier,
+                                                     @Nonnull Class<T> clazz, int chunkSize) {
+        checkTrue(query != null, "query cannot be null");
+        checkTrue(database != null, "database cannot be null");
+        checkTrue(connectionSupplier != null, "username cannot be null");
+        checkTrue(clazz != null, "clazz cannot be null");
+
+        return SourceBuilder.stream("influxdb-" + database,
+                ignored -> new InfluxDbStreamingSource<>(query, database, chunkSize, connectionSupplier, clazz))
+                .<T>fillBufferFn(InfluxDbStreamingSource::addToBufferWithMapping)
+                .destroyFn(InfluxDbStreamingSource::close)
+                .build();
+    }
+
+    /**
+     * A streaming source which executes a streaming query on InfluxDb and emits
+     * results as they arrive.
+     *
+     * @param <T> emitted item type
+     */
+    private static class InfluxDbStreamingSource<T> {
+
+        private final Class<T> clazz;
+        private final BlockingQueue<QueryResult> queue = new LinkedBlockingQueue<>(10000);
+        private final ArrayList<QueryResult> buffer = new ArrayList<>();
+        private final InfluxDBResultMapper resultMapper;
+        private InfluxDB db;
+        private volatile boolean finished;
+
+        InfluxDbStreamingSource(@Nonnull String query, @Nonnull String database, int chunkSize,
+                                @Nonnull SupplierEx<InfluxDB> connectionSupplier, @Nullable Class<T> clazz) {
+            this.clazz = clazz;
+            this.resultMapper = clazz != null ? new InfluxDBResultMapper() : null;
+            db = connectionSupplier.get();
+            db.query(new Query(query, database),
+                    chunkSize,
+                    queue::add,
+                    () -> finished = true
+            );
+        }
+
+        void addToBufferWithMapping(SourceBuffer<T> sourceBuffer) {
+            checkTrue(resultMapper != null, "resultMapper cannot be null!");
+            checkTrue(clazz != null, "clazz cannot be null!");
+            transferTo(result -> {
+                if (!result.hasError()) {
+                    resultMapper.toPOJO(result, clazz).forEach(sourceBuffer::add);
+                }
+            }, sourceBuffer);
+        }
+
+        void addToBufferWithOutMapping(SourceBuffer<Series> sourceBuffer) {
+            transferTo(result -> addResultsToSourceBuffer(result, sourceBuffer), sourceBuffer);
+        }
+
+        private <B> void transferTo(Consumer<QueryResult> consumer, SourceBuffer<B> sourceBuffer) {
+            queue.drainTo(buffer);
+            buffer.forEach(consumer);
+            buffer.clear();
+            if (finished && queue.isEmpty()) {
+                sourceBuffer.close();
+            }
+        }
+
+        void close() {
+            if (db != null) {
+                db.close();
+                db = null;
```

Comment:
we don't need to do null assignment here, the processor will be cleaned up automatically